src/field_theory/adjoin.lean

```diff
@@ -25,29 +25,88 @@ For example, `algebra.adjoin K {x}` might not include `x⁻¹`.
  - `F⟮α⟯`: adjoin a single element `α` to `F`.
 -/
 
-namespace field
+namespace intermediate_field
 
 section adjoin_def
 variables (F : Type*) [field F] {E : Type*} [field E] [algebra F E] (S : set E)
 
 /-- `adjoin F S` extends a field `F` by adjoining a set `S ⊆ E`. -/
-def adjoin : subalgebra F E :=
-{ carrier := field.closure (set.range (algebra_map F E) ∪ S),
-  one_mem' := is_submonoid.one_mem,
-  mul_mem' := λ x y, is_submonoid.mul_mem,
-  zero_mem' := is_add_submonoid.zero_mem,
-  add_mem' := λ x y, is_add_submonoid.add_mem,
-  algebra_map_mem' := λ x, field.mem_closure (or.inl (set.mem_range.mpr ⟨x,rfl⟩)) }
+def adjoin : intermediate_field F E :=
+{ algebra_map_mem' := λ x, subfield.subset_closure (or.inl (set.mem_range_self x)),
+  ..subfield.closure (set.range (algebra_map F E) ∪ S) }
+
+lemma adjoin_le_iff {T : intermediate_field F E} : adjoin F S ≤ T ↔ S ≤ T :=
+⟨λ H, le_trans (le_trans (set.subset_union_right _ _) subfield.subset_closure) H,
+  λ H, (@subfield.closure_le E _ (set.range (algebra_map F E) ∪ S) T.to_subfield).mpr
+  (set.union_subset (intermediate_field.set_range_subset T) H)⟩
+
+end adjoin_def
+
+section lattice
+variables {F : Type*} [field F] {E : Type*} [field E] [algebra F E]
+
+lemma gc : galois_connection (adjoin F : set E → intermediate_field F E) coe := adjoin_le_iff F
+
+/-- Galois insertion between `adjoin` and `coe`. -/
+def gi : galois_insertion (adjoin F : set E → intermediate_field F E) coe :=
+{ choice := λ S _, adjoin F S,
+  gc := intermediate_field.gc,
+  le_l_u := λ S, (intermediate_field.gc (S : set E) (adjoin F S)).1 $ le_refl _,
+  choice_eq := λ _ _, rfl }
+
+instance : complete_lattice (intermediate_field F E) :=
+galois_insertion.lift_complete_lattice intermediate_field.gi
+
+instance : inhabited (intermediate_field F E) := ⟨⊤⟩
+
+lemma mem_bot {x : E} : x ∈ (⊥ : intermediate_field F E) ↔ x ∈ set.range (algebra_map F E) :=
+begin
+  suffices : set.range (algebra_map F E) = (⊥ : intermediate_field F E),
+  { rw this, refl },
+  { change set.range (algebra_map F E) = subfield.closure (set.range (algebra_map F E) ∪ ∅),
+    simp [←set.image_univ, ←ring_hom.map_field_closure] }
+end
+
+lemma mem_top {x : E} : x ∈ (⊤ : intermediate_field F E) :=
+subfield.subset_closure $ or.inr trivial
+
+lemma bot_to_subalgebra_eq_bot : (⊥ : intermediate_field F E).to_subalgebra = ⊥ :=
+begin
+  ext,
+  rw [mem_to_subalgebra, algebra.mem_bot, mem_bot],
+end
+
+lemma top_to_subalgebra_eq_top : (⊤ : intermediate_field F E).to_subalgebra = ⊤ :=
+begin
+  ext,
+  rw [mem_to_subalgebra, iff_true_right algebra.mem_top],
+  exact mem_top,
+end
+
+lemma lift2_bot (K : intermediate_field F E) : ↑(⊥ : intermediate_field K E) = K :=
```

Comment:
What is this `lift2`? Is it just `coe`?
This looks like a simp lemma.

Reply:
lift2 is a bad name, but basically there are two different lifts from intermediate field to intermediate field. If E/K/F then there is K/F -> E/F and E/K -> E/F. lift1 is what I called the first one, and lift2 is what I called the second one, although there are  probably better names for both the lifts and these lemmas.

Reply:
The issue is that eventually we will also have a lemma that `lift1 (bot : K/F) = (bot : E/F)`, and these two lemmas can't both be called coe_bot.

Reply:
I think we have something similar for `(sub)algebra`s. You could check what happens there. Maybe @Vierkantor or @kckennylau knows?

I guess this one could be called `coe_bot_eq_self`?

Reply:
The equivalent for `subalgebra` is called [`subalgebra.res`](https://github.com/leanprover-community/mathlib/blob/master/src/ring_theory/algebra_tower.lean#L235) by the way.

Reply:
Does subalgebra have a name for the other type of lift?